server/Main.hs

```diff
@@ -71,10 +71,14 @@ server externs initNamesEnv initEnv port = do
           case CST.parseModuleFromFile "<file>" input >>= CST.resFull of
             Left parseError ->
               return . Left . CompilerErrors . P.toJSONErrors False P.Error $ CST.toMultipleErrors "<file>" parseError
-            Right m | P.getModuleName m == P.ModuleName "Main" -> do
+            Right m -> do
+              -- Rewrite the module name to "$Main" in order to ensure that the
+              -- module name doesn't clash with any existing module names in
+              -- the package set.
+              let rewriteModuleName (P.Module ss coms _ decls refs) = P.Module ss coms (P.moduleNameFromString "$Main") decls refs
```

Comment:
If someone attempts to play with module export syntax this might be cause problems with:
```
module Foo (module Bar, module Foo) where ...
```

Reply:
I'll wager this is by far an edge case, however 😆

Reply:
Ah, good point. I think re-exporting `main` from elsewhere seems plausible to want to do, so perhaps this isn't the best approach. Maybe we should assemble a set of used module names from the externs files, and then instead of rewriting, just check that the given module name is not included in that set?